### PR TITLE
Draft: Chinese text view

### PR DESCRIPTION
**Pull request by sebastian-nehrdich**

### CODE REVIEW

File: src/views/textview/text-view-right.js [hunk 1]
```diff
@@ -81,6 +82,22 @@ export class TextViewRight extends LitElement {
...
     });
   }
+  scrollAfterEndlessReload() {
+    if (this.noScrolling && this.EndlessScrollFlag) {
+      if (this.activeSegment) {
+        let activeElement = this.shadowRoot.getElementById(this.activeSegment);
+        let mainScrollPosition = this.scrollTop;
+        if (this.currentPosition > 100) {
+          activeElement.scrollIntoView({ block: 'end', inline: 'nearest' });
+          this.scrollTop += 0;
```
Comment (ayya-vimala) [inline src/views/textview/text-view-right.js]: This seems pretty useless.

File: src/views/textview/text-view-right.js [hunk 2]
```diff
@@ -81,6 +82,22 @@ export class TextViewRight extends LitElement {
...
+        } else {
+          activeElement.scrollIntoView({ block: 'start', inline: 'nearest' });
+          this.scrollTop -= 0;
```
Comment (ayya-vimala) [inline src/views/textview/text-view-right.js]: ditto

File: src/views/visual/visual-view-graph.js [hunk 3]
```diff
@@ -134,11 +158,12 @@ export class VisualViewGraph extends LitElement {
       ? searchTerm.split('_')[1]
       : searchTerm.replace('tib_', '');
     console.log('visual view: fetching data', this.searchItem);
-    const { graphdata, error } = await getDataForVisual({
+    let { graphdata, error } = await getDataForVisual({
       searchTerm: searchTerm,
       selected: this.selectedCollections,
       language: this.language,
     });
+    graphdata = this.preprocessGraphData(graphdata);
```
Comment (ayya-vimala) [inline src/views/visual/visual-view-graph.js]: best not use this kind of constructions that change variables in this way. Best to use const and then use another name for the output.

File: src/views/visual/visual-view-graph.js [hunk 4]
```diff
@@ -119,6 +119,30 @@ export class VisualViewGraph extends LitElement {
       </div>
     `;
   }
+  // the following could also be done on the backend, but I am not entirely sure. it is a complex question...
```
Comment (ayya-vimala) [inline src/views/visual/visual-view-graph.js]: Indeed best done in the backend.
But why delete the numbers on the right hand side? Very difficult for me to see which collection I'm looking at.
What are you trying to achieve? I notice that on this branch in local it is different from buddhanexus.net

Comment (sebastian-nehrdich) [reply src/views/visual/visual-view-graph.js]: it reduces the number of collections and makes the display managable for the Chinese. In the Chinese canon (contrary to the Tibetan), for example Abhidharma is split into T26,T27,T28 etc. The plan here is to just sum them up into larger baskets, which means less nodes on the visualization and a cleaner display.

Comment (sebastian-nehrdich) [reply src/views/visual/visual-view-graph.js]: however, I believe that on the level of data-organization, we better stick to the layout of the CBETA and just do this for the visualization. I played a bit on the backend to make these changes already when reading in the dataset but that's a hopeless (= horrifying) undertaking and I don't want to try that again. It introduces all kinds of problems. But we can probably do this in python after we receive the data from the query and before we dispatch it to the API.

**Pull request merged**